src/irmin-pack/inode.ml

```diff
@@ -530,94 +586,111 @@ struct
           let i = index ~depth s in
           match entries.(i) with
           | None ->
-              let target = values (StepMap.singleton s v) in
-              entries.(i) <- entry ~target target.hash;
-              let t = tree { depth; length; entries } in
+              let target = values layout (StepMap.singleton s v) in
+              entries.(i) <- Some (Ptr.of_target layout target);
```

Comment:
not sure it's worth it, but you could extract the `Ptr.of_target layout` out of the recursive function

Reply:
Not sure it's worth it, yes.